tests/src/test/scala/com/evolutiongaming/skafka/FiberWithBlockingCancel.scala

```diff
@@ -0,0 +1,49 @@
+package com.evolutiongaming.skafka
+
+import cats.implicits._
+import cats.effect.syntax.all._
+import cats.effect.concurrent.Deferred
+import cats.effect.{Concurrent, Fiber, Resource, Timer}
+
+import scala.concurrent.duration.FiniteDuration
+
+object FiberWithBlockingCancel {
```

Comment:
should we move this to cats-helper project?

Reply:
not yet, I would keep it as part of this PR for now, also I think that we might wanna introduce async boundary at
```
fiber <- self.guarantee {
```
to avoid issues like https://github.com/typelevel/cats-effect/issues/326#issuecomment-416925044

Reply:
created https://github.com/evolution-gaming/skafka/issues/129